### PR TITLE
Feature/message converter

### PR DESCRIPTION
**Pull request by ignacio83**
Let me know, what you think about....

### CODE REVIEW

File: src/main/java/net/reini/rabbitmq/cdi/AbstractMessageConverter.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    public byte[] toBytes(Object object) throws MessageConverterException{
+        try {
+            return toBytesInner(object);
+        } catch (Exception e) {
+            throw new MessageConverterException(e);
+        }
+    }
+    
+    public abstract byte[] toBytesInner(Object object) throws Exception;
```
Comment (reinhapa) [inline src/main/java/net/reini/rabbitmq/cdi/AbstractMessageConverter.java]: What is the reason for this abstract implementation exactly when we have a `MessageConverter` interface, where we could add a default method for the content type for example...

File: src/main/java/net/reini/rabbitmq/cdi/EventBinder.java [hunk 2]
```diff
@@ -297,6 +300,18 @@ public ExchangeBinding withRoutingKey(String routingKey) {
...
       return this;
     }
+    
+    /**
+     * Sets the message converter to be user for message publishing.
+     * 
+     * @param messageConverter The message converter 
+     * @return the exchange binding
+     */
+    public ExchangeBinding withConverter(MessageConverter messageConverter){
```
Comment (reinhapa) [inline src/main/java/net/reini/rabbitmq/cdi/EventBinder.java]: Here we should convert the `ExchangeBinding` to a generic variant of `bind(Class<M> messageType)` to allow verification of the generic `MessageConverter<M>`.  This I would rename to `MessageEncoder<M>`  to have `MessageDecoder<M>` the receiving end.

File: src/main/java/net/reini/rabbitmq/cdi/MessageConverter.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package net.reini.rabbitmq.cdi;
+
+/**
+ * Transform a object in a array de bytes to send a message.
+ * 
+ * @author André Ignacio
+ */
+public interface MessageConverter {
```
Comment (reinhapa) [inline src/main/java/net/reini/rabbitmq/cdi/MessageConverter.java]: I would like to have a generic type `M` of a converter like: `public interface MessageConverter<M>` for better type savety.

File: src/main/java/net/reini/rabbitmq/cdi/MessageConverter.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  /**
+   * Transform a object in a array de bytes to send a message.
+   *  
+   * @return Array of bytes
+   * @throws MessageConverterException If the conversion fails
+   */
+  byte[] toBytes(Object object) throws MessageConverterException;
```
Comment (reinhapa) [inline src/main/java/net/reini/rabbitmq/cdi/MessageConverter.java]: Here we would better use `byte[] encode(M message) throws MessageConverterException`

File: src/main/java/net/reini/rabbitmq/cdi/MessageConverterException.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
+package net.reini.rabbitmq.cdi;
+
+/**
+ * Exception for failures in message conversion.
+ * 
+ * @author André Ignacio 
+ */
+public class MessageConverterException extends RuntimeException {
```
Comment (reinhapa) [inline src/main/java/net/reini/rabbitmq/cdi/MessageConverterException.java]: In order to use it on a `MessageEncoder` or `MessageDecoder` we should rename it to `MessageEncodingException`

**Comment by reinhapa** [comment]
Please do also format your classes using the Google Format

**Comment by ignacio83** [comment]
Done.

**Comment by reinhapa** [comment]
Thanks for the changes

**Pull request merged**